storage/atsd/atsd.go

```diff
@@ -0,0 +1,400 @@
+	protocol        = flag.String("storage_driver_atsd_write_protocol", "", "transfer protocol. Possible settings: http/https, udp, tcp")
+	connectionLimit = flag.Uint("storage_driver_atsd_connection_limit", 1, "ATSD storage driver TCP connection count")
+	memstoreLimit   = flag.Uint64("storage_driver_atsd_buffer_limit", 1000000, "maximum number of series commands stored in buffer until flush")
+
+	configFilePath = flag.String("storage_driver_atsd_config_path", "", "path to ATSD storage driver config file")
+
+	dockerHost             = flag.String("storage_driver_atsd_docker_host", "docker-host", "hostname of machine where docker daemon is running")
+	includeAllMajorNumbers = flag.Bool("storage_driver_atsd_store_major_numbers", false, "include statistics for devices with all available major numbers")
+	userCgroupsEnabled     = flag.Bool("storage_driver_atsd_store_user_cgroups", false, "include statistics for \"user\" cgroups (for example: docker-host/user.*)")
+	propertyInterval       = flag.Duration("storage_driver_atsd_property_interval", 1*time.Minute, "container property update interval. Should be >= housekeeping_interval")
+	samplingInterval       = flag.Duration("storage_driver_atsd_sampling_interval", *manager.HousekeepingInterval, "series sampling interval. Should be >= housekeeping_interval")
+	deduplication          = deduplicationParamsList(map[string]atsdStorageDriver.DeduplicationParams{})
+)
+
+func init() {
+	storage.RegisterStorageDriver("atsd", new)
+
+	flag.Var(&deduplication, "storage_driver_atsd_deduplication",
+		"Specify optional deduplication settings for a metric group using 'group:interval:threshold' syntax. "+
+			"Group - Metric group to which the setting applies. Supported metric groups in cAdvisor: cpu, memory, io, network, task, filesystem. "+
+			"Interval - Maximum delay between the current and previously sent samples. If exceeded, the current sample is sent to ATSD regardless of the specified threshold. "+
+			"Threshold - Absolute or percentage difference between the current and previously sent sample values. If the absolute difference is within the threshold and elapsed time is within Interval, the value is discarded.")
+
+	//these parameters are bounded below by a housekeeping interval
+	if *propertyInterval < *manager.HousekeepingInterval {
+		*propertyInterval = *manager.HousekeepingInterval
+	}
+	if *samplingInterval < *manager.HousekeepingInterval {
+		*samplingInterval = *manager.HousekeepingInterval
+	}
+}
+
+func new() (storage.StorageDriver, error) {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
+	return newStorage(
+		*storage.ArgDbUsername,
+		*storage.ArgDbPassword,
+		hostname,
+		*storage.ArgDbBufferDuration,
+	), nil
+}
+
+type deduplicationParamsList map[string]atsdStorageDriver.DeduplicationParams
+
+func (self deduplicationParamsList) String() string {
+	m := map[string]atsdStorageDriver.DeduplicationParams(self)
```

Comment:
I meant is the cast necessary? Does the `Sprint` function format it any differently without the cast?